atomic_long_it_test.go

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast_test
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+func TestAtomicLong(t *testing.T) {
+	testCases := []struct {
+		name       string
+		f          func(t *testing.T)
+		noParallel bool
+	}{
+		{name: "AtomicLongAddAndGet", f: atomicLongAddAndGetTest},
+		{name: "AtomicLongAlter", f: atomicLongAlterTest},
+		{name: "AtomicLongAlterAndGet", f: atomicLongAlterAndGetTest},
+		{name: "AtomicLongApply", f: atomicLongApplyTest},
+		{name: "AtomicLongCompareAndSet_Fail", f: atomicLongCompareAndSetFailTest},
+		{name: "AtomicLongCompareAndSet_Success", f: atomicLongCompareAndSetSuccessTest},
+		{name: "AtomicLongDecrementAndGet", f: atomicLongDecrementAndGetTest},
+		{name: "AtomicLongGet", f: atomicLongGetTest},
+		{name: "AtomicLongGetAndAdd", f: atomicLongGetAndAddTest},
+		{name: "AtomicLongGetAndAlter", f: atomicLongGetAndAlterTest},
+		{name: "AtomicLongGetAndDecrement", f: atomicLongGetAndDecrementTest},
+		{name: "AtomicLongGetAndIncrement", f: atomicLongGetAndIncrementTest},
+		{name: "AtomicLongGetAndSet", f: atomicLongGetAndSetTest},
+		{name: "AtomicLongIncrementAndGet", f: atomicLongIncrementAndGetTest},
+		{name: "AtomicLongSet", f: atomicLongSetTest},
+	}
+	// run no-parallel test first
+	sort.Slice(testCases, func(i, j int) bool {
+		return testCases[i].noParallel && !testCases[j].noParallel
+	})
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			if !tc.noParallel {
+				t.Parallel()
+			}
+			tc.f(t)
+		})
+	}
+}
+
+// ported from: com.hazelcast.cp.internal.datastructures.atomiclong.AbstractAtomicLongBasicTest#testSet
```

Comment:
Could you move thie "ported from:" comments after the function definition? Otherwise these become the function documentation:
```
func atomicLongSetTest( ...) {
// ported from: com.hazelcast.cp.internal.datastructures.atomiclong.AbstractAtomicLongBasicTest#testSet
```

Reply:
Fixed at https://github.com/hazelcast/hazelcast-go-client/pull/901/commits/3604ce1f707a5e723a8c0241ba1108ccfad17b67.